### ISSUE TITLE
rpc error: rpc: can't find method 

### ISSUE THREAD

**Issue by bvanhou** [hashicorp/nomad#9972]
New Issue: Needing assistance from Hashicorp identifying customer issue

### Nomad version
Version: 0.6.0-dev

### Operating system and Environment details
CircleCI Enterprise Server Environment Hosted on AWS

### Issue
Nomad service cannot build job.

### Reproduction steps
Unable to establish exact reproduction steps, we believe it is due to some kind of network configuration issue. However, error occurs on customer environment inconsistently. 
Need help identifying more information around the Nomad service error to help determine where the error is coming from.

### Nomad Client logs (if appropriate)
```
Submit Date   = <none>
Type          = batch
Priority      = 50
Datacenters   = us-east-1
Status        = pending
Periodic      = false
Parameterized = false
Error querying latest job deployment: Unexpected response code: 500 (1 error(s) occurred:
* RPC failed to server 172.30.34.4:4647: rpc error: rpc: can't find method Job.LatestDeployment)
```

**Comment by tgross** [comment]
@bvanhou can you confirm that all clients and servers are on the same version of Nomad? Also, generally speaking this version of Nomad is out of support. We strongly recommend you upgrade the cluster to a supported version.

**Comment by bvanhou** [comment]
The services box and the clients are all on Version: 0.6.0-dev. The next steps for me is to get them to the next version of nomad. I will keep you posted on the results.

**Comment by tgross** [comment]
The `Job.LatestDeployment` RPC does appear in 0.6.0 (see [`job_endpoint.go#L826`](https://github.com/hashicorp/nomad/blob/v0.6.0/nomad/job_endpoint.go#L826)), but that was introduced in d0a83321403a33cb9953c578272d620d2bbf6e64. Because you've got Nomad version 0.6.0-dev, it's not likely that it includes d0a83321403a33cb9953c578272d620d2bbf6e64 which shipped in the 0.6.0-rc1 release candidate.
